File: src/week1/P001.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0], 10);
const numList = input.slice(1).map(Number);

function merge(left, right) {
    const result = [];
    let i = 0;
    let j = 0;
    while (i < left.length && j < right.length) {
        if (left[i] <= right[j]) {
            result.push(left[i]);
            i++;
        } else {
            result.push(right[j]);
            j++;
        }
    }

    return result.concat(left.slice(i), right.slice(j));
}

function mergeSort(arr) {
    if (arr.length <= 1) {
        return arr;
    }
    
    // 1. 분할 (Divide)
    const mid = Math.floor(arr.length / 2);
    const left = arr.slice(0, mid);
    const right = arr.slice(mid);

    // 2. 정복 (Conquer) & 결합 (Combine)
    return merge(mergeSort(left), mergeSort(right));
}

const sortedList = mergeSort(numList);

console.log(sortedList.join('\n'));

File: src/week1/P001.py
# 2750
n = int(input())
result = []
for i in range(n):
    result.append(int(input()))
result.sort()

for i in range(n):
    print(result[i])

## 병합 정렬(merge sort)

import sys

def merge(left, right):
    result = []
    i, j = 0, 0
    while i < len(left) and j < len(right):
        if left[i] <= right[j]:
            result.append(left[i])
            i += 1
        else:
            result.append(right[j])
            j += 1

    result.extend(left[i:])
    result.extend(right[j:])
    return result

def merge_sort(arr):
    if len(arr) <= 1:
        return arr
    
    mid = len(arr) // 2
    left = arr[:mid]
    right = arr[mid:]

    return merge(merge_sort(left), merge_sort(right))

n = int(input())
numList = []

for i in range(n):
    numList.append(int(input()))

result = merge_sort(numList)

for nums in result:
    print(nums)